changes/924.deprecation.md

```diff
@@ -0,0 +1,3 @@
+Deprecated `RESTClient.command_builder` and `RESTClient.create_application_command`.
+
+Instead, you are encouraged to switch to `RESTClient.slash_command_builder`/`RESTClient.context_menu_builder` and `RESTClient.create_slash_command`/`RESTClient.create_context_menu_command` respectively.
```

Comment:
Encouraged is the wrong word here, you should switch its a deprecation not a "this new method is preferred"